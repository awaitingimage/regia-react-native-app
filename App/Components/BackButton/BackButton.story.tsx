import { storiesOf } from "@storybook/react-native";
import * as React from "react";

import BackButton from "./BackButton";

storiesOf("BackButton")
  .add("Default", () => (
    <BackButton />
  ))
  .add("Black", () => (
    <BackButton color={"black"} />
  ));
