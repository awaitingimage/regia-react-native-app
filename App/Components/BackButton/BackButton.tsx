import * as React from "react";
import { GestureResponderEvent, StyleProp, TouchableOpacity, ViewStyle } from "react-native";
import Icon from "react-native-vector-icons/Ionicons";
import styles from "./Style";

interface Props {
  style?: StyleProp<ViewStyle>;
  onPress?: (event: GestureResponderEvent) => void;
  color?: string;
}

const defaultProps = {
  style: styles.container,
  onPress: () => alert("back button"),
  color: "black",
};

const BackButton: React.SFC<Props> = ({style, onPress, color}: Props) =>  (
  <TouchableOpacity accessibilityLabel={"back"} onPress={onPress} style={style}>
    <Icon name="ios-arrow-back" style={[styles.icon, {color}]} />
  </TouchableOpacity>
);

BackButton.defaultProps = defaultProps;

export default BackButton;
