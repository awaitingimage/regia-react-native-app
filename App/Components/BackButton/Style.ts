import { StyleSheet } from "react-native";
import Metrics from "../../Themes/Metrics";

export default StyleSheet.create({
  container: {
    position: "absolute",
    left: Metrics.baseMargin,
    top: Metrics.baseMargin,
  },
  icon: {
    padding: Metrics.baseMargin,
    fontSize: 32,
  },
});
