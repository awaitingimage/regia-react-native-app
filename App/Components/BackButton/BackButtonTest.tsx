/// <reference types="@types/jest" />
import * as React from "react";
import * as renderer from "react-test-renderer";
import BackButton from "./BackButton";

test("BackButton component renders correctly", () => {
  const tree = renderer.create(<BackButton />).toJSON();
  expect(tree).toMatchSnapshot();
});
