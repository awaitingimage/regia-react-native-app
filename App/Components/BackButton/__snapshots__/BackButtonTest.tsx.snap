// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton component renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel="back"
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "left": 10,
      "opacity": 1,
      "position": "absolute",
      "top": 10,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "fontSize": 32,
            "padding": 10,
          },
          Object {
            "color": "black",
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
</View>
`;
